src/edu/ccsu/cs505/hw2/BankAccount.java

```diff
@@ -0,0 +1,84 @@
+package edu.ccsu.cs505.hw2;
+
+/**
+ * @author Julie Shevchenko
+ * BankAccount class, which extends Account.
+ * Overrides setNextChain and deduct methods 
+ **/
+public class BankAccount extends Account{
+    private String accountName;
+    private double accountBalance;
+    private Account chainAccount;
```

Comment:
You have the next account in the abstract account class, you shouldn't have it in the child class as well.